tests/src/test/java/org/eclipse/hono/tests/coap/CoapTestBase.java

```diff
@@ -1053,6 +1059,53 @@ private URI getRequestUri(final String scheme, final String resource) {
         }
     }
 
+    /**
+     * Creates a CoAP request using either the <em>coaps</em> or <em>coap</em> scheme,
+     * depending on the given endpoint configuration.
+     *
+     * @param endpointConfig The endpoint configuration.
+     * @param requestDeviceId The identifier of the device to publish data for.
+     * @param msgNo The message number.
+     * @return The request to send.
+     */
+    protected Request createCoapsOrCoapRequest(
+            final CoapCommandEndpointConfiguration endpointConfig,
+            final String requestDeviceId,
+            final int msgNo) {
+
+        if (endpointConfig.isSubscribeAsGatewayForSingleDevice()) {
+            return createCoapsRequest(Code.PUT, getMessageType(), getPutResource(tenantId, requestDeviceId), msgNo);
+        } else if (endpointConfig.isSubscribeAsUnauthenticatedDevice()) {
+            return createCoapRequest(Code.PUT, getMessageType(), getPutResource(tenantId, requestDeviceId), msgNo);
+        } else {
+            return createCoapsRequest(Code.POST, getMessageType(), getPostResource(), msgNo);
```

Comment:
I'm not sure, if a device, which doesn't use coaps, could be considered to be authenticated?

Reply:
The not-using-coaps request handling is done in the `UnauthenticatedDevice` block.